### PR TITLE
Fix filter clicks on array-valued JSON fields

### LINKED ISSUE #179: Filter clicks on array-valued JSON fields produce incorrect queries
## Problem

When clicking the filter icon on an array-valued field in the Grafana log detail panel (e.g. `attributes.tags = ["paperclip"]`), the plugin generates a phrase query that matches no documents.

The value comes through as the stringified array `["paperclip"]`. The `=` operator wraps it in quotes producing `attributes.tags:"[\"paperclip\"]"`, which searches for the literal string including brackets. The indexed terms are the individual array elements (`paperclip`), not the JSON array representation.

## Expected behavior

Clicking the filter icon on `attributes.tags: ["paperclip"]` should generate `attributes.tags:paperclip` (a term query matching the array element).

## Root cause

Two issues in `addFilterToQuery` (src/datasource/base.ts):

1. **Operator selection**: When `fieldTypes` isn't populated (race condition on datasource init) or the field is `keyword` type, the `=` operator is chosen instead of `term`. The `=` path generates phrase queries via `LuceneQuery.addFilter`.

2. **Value unwrapping**: The value `["paperclip"]` needs to be detected as a JSON array and unwrapped to its element(s). For multi-element arrays, this could generate an OR of term queries.

## Environment

- Quickwit OTEL log index with `attributes` as a JSON field (`tokenizer: "raw"`, `record: "basic"`)
- All JSON subfields report as both `keyword` and `text` via `_field_caps`

### PR DESCRIPTION
**Pull request by xrl**
## Summary
- When Grafana passes a stringified JSON array (e.g. `["paperclip"]`) as a filter value, the `=` operator wrapped it in quotes producing `attributes.tags:"[\"paperclip\"]"` — a phrase query matching no documents
- Detects JSON array values and unwraps them into term queries on individual elements
- Single element: `attributes.tags:paperclip`
- Multiple elements: `attributes.tags:IN ["paperclip" "stapler"]` (tantivy TermSetQuery)
- Empty arrays are a no-op
- Non-array values starting with `[` that aren't valid JSON pass through unchanged

Note: tantivy indexes array elements as individual terms — there's no way to match on array length, order, or exact composition. For multi-element arrays we use IN (match any), which is the most useful behavior for log exploration filters.

## Test plan
- [x] 18 unit tests covering all ad-hoc filter operators and array edge cases
- [ ] Manual test: click filter icon on an array field in Grafana log detail, verify results appear

Fixes #179

### CODE REVIEW

File: src/modifyQuery.ts [hunk 1]
```diff
@@ -18,6 +33,24 @@ export function addAddHocFilter(query: string, filter: AdHocVariableFilter): str
...
+    // which is the most useful behavior for log exploration filters.
+    const arrayElements = tryParseJsonArray(filter.value);
+    if (arrayElements !== null) {
+      if (arrayElements.length === 0) {
+        return query;
+      }
+      const modifier = filter.operator === '=' ? '' : '-';
+      const key = escapeFilter(filter.key);
+      if (arrayElements.length === 1) {
+        return concatenate(query, `${modifier}${key}:${escapeFilterValue(arrayElements[0])}`, 'AND');
```
Comment (Sh4d1) [inline src/modifyQuery.ts]: Do we want to support values like `["foo bar"]` and `["foo:bar"]` ? 
Cause these will break here

Comment (xrl) [reply src/modifyQuery.ts]: Great catch, done! Thanks

**Comment by xrl** [comment]
@Sh4d1 ready for another review

**Comment by Sh4d1** [comment]
Thanks @xrl !
